x-pack/packages/kbn-cloud-security-posture/common/utils/helpers.ts

```diff
@@ -62,3 +62,43 @@ export const buildEntityFlyoutPreviewQuery = (field: string, queryValue?: string
     },
   };
 };
+
+export const buildEntityAlertsQuery = (field: string, queryValue?: string, size?: number) => {
```

Comment:
I don't like the fact that we use our own logic of querying alerts for preview, even though the component should have exactly the same logic as the alert preview component in the document flyout. I'd push for finding a way to use the same query (not identical, but literally the same code) so when the logic changes it changes only in one place

Reply:
So I ran into an issue trying to use filter when using the same logic as alerts preview component in document flyout, ( I discussed this with @christineweng  , and we thought its best to use different logic for now 

currently the logic used in document flyout didn't exclude 'closed' alerts

Reply:
it's hard for me to reason about it not knowing what the problems were. Excluding `closed` alerts from the query shouldn't be a big problem (ofc I might miss some hidden complexity). But anyway we need to change the current logic because we don't want to load documents each time preview is shown plus we don't want to be limited by 500 alerts, so we only need the aggregation. When you switch to the aggregation, then I'd give it another go to try and have the shared logic around the query, if the logic is exactly the same and it's in similar component (flyout), I don't see a good reason not to have the same query logic yet

Reply:
I have updated the logic to use different hooks now, it is still currently different than what the alerts preview on document flyout is using, but I have discussed with @christineweng about this and we agree that the document flyout can also use this same logic

Reply:
@animehart do we still need this `buildEntityAlertsQuery`? It's not used anywhere on this pr. if it's needed for the next one, let's add it there